scene/2d/navigation_agent_2d.cpp

```diff
@@ -314,6 +321,8 @@ Vector2 NavigationAgent2D::get_target_location() const {
 
 Vector2 NavigationAgent2D::get_next_location() {
 	update_navigation();
+
+	const Vector<Vector2> navigation_path = navigation_result->get_path();
```

Comment:
I would use a const reference here unless you intend to modify:
```
const Vector<Vector2> &navigation_path
```
The COW probably reduces most of the cost but it is better to be explicit, there's no need for all that machinery just to effectively get the size of the path.

Reply:
These have been changed to references.